specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/EasyAuthConfigs.json

```diff
@@ -408,6 +408,102 @@
         "azureStaticWebApp": {
           "$ref": "#/definitions/AzureStaticWebApp",
           "description": "The configuration settings of the Azure Static Web Apps provider."
+        },
+        "customOpenIdConnectProviders": {
+          "description": "The map of the name of the alias of each custom Open ID Connect provider to the\nconfiguration settings of the custom Open ID Connect provider.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/CustomOpenIdConnectProvider"
+          }
+        }
+      }
+    },
+    "CustomOpenIdConnectProvider": {
+      "description": "The configuration settings of the custom Open ID Connect provider.",
+      "type": "object",
+      "properties": {
+        "state": {
+          "$ref": "#/definitions/IdentityProviderState",
+          "description": "<code>Disabled</code> if the custom Open ID Connect provider provider should not be enabled despite the set registration; otherwise, <code>Enabled</code>."
+        },
+        "registration": {
+          "$ref": "#/definitions/OpenIdConnectRegistration",
+          "description": "The configuration settings of the app registration for the custom Open ID Connect provider."
+        },
+        "login": {
+          "$ref": "#/definitions/OpenIdConnectLogin",
+          "description": "The configuration settings of the login flow of the custom Open ID Connect provider."
+        }
+      }
+    },
+    "OpenIdConnectRegistration": {
+      "description": "The configuration settings of the app registration for the custom Open ID Connect provider.",
+      "type": "object",
+      "properties": {
+        "clientId": {
+          "description": "The client id of the custom Open ID Connect provider.",
+          "type": "string"
+        },
+        "clientCredential": {
+          "$ref": "#/definitions/OpenIdConnectClientCredential",
+          "description": "The authentication credentials of the custom Open ID Connect provider."
+        },
+        "openIdConnectConfiguration": {
+          "$ref": "#/definitions/OpenIdConnectConfig",
+          "description": "The configuration settings of the endpoints used for the custom Open ID Connect provider."
+        }
+      }
+    },
+    "OpenIdConnectClientCredential": {
+      "description": "The authentication client credentials of the custom Open ID Connect provider.",
+      "type": "object",
+      "properties": {
+        "clientSecretRefName": {
```

Comment:
Suggest to add `x-ms-secret`: true for secrets.

Reply:
It is just secret name instead of actual secret, the actual secret is saved in other place